src/ui/public/visualize/loader/visualize_loader.js

```diff
@@ -52,46 +51,44 @@ import { EmbeddedVisualizeHandler } from './embedded_visualize_handler';
  * @property {object} query The query that should apply to that visualization.
  */
 
-const VisualizeLoaderProvider = ($compile, $rootScope, savedVisualizations) => {
-  const renderVis = (el, savedObj, params) => {
-    const scope = $rootScope.$new();
-    params = params || {};
-    scope.savedObj = savedObj;
-    scope.uiState = params.uiState;
-    scope.timeRange = params.timeRange;
-    scope.filters = params.filters;
-    scope.query = params.query;
-    scope.updateState = (visState) => {
-      if (params.appState) {
-        params.appState.vis = visState;
-        params.appState.save();
-      }
-    };
+const VisualizeLoaderProvider = ($compile, $rootScope, savedVisualizations, Private) => {
+  const renderVis = (container, savedObj, params) => {
 
-    const container = angular.element(el);
+    const { vis, description } = savedObj;
+    // TODO: where is this even set, and could it not be set directly on the vis object ?
```

Comment:
Do you want to address this TODO in this PR? If not, would it make sense to create an issue for this? Without context, this seems like the kind of comment that might be more confusing than helpful to someone who comes across it in a month or two.